installation issue on AMD 7551 with gcc@8.2.0 - illegal instruction

I'm getting "illegal instruction" when trying to use spack to install just about anything on an AMD EPYC 7511 "excavator" / Naples system (Azure HB60rs instance) with a vanilla CentOS 7.6 image. The same process is working fine on it's sister Intel Skylake platform. 

The example given is gcc@8.2.0, which fails in perl. Other packages are failing with the same illegal instruction error (e.g. libatomic). When building the packages by hand outside of spack, they work fine. The offending illegal instruction appears to be BMI1 "bextr" which this platform does appear to support according to the cpuinfo output below.

Spack is using only the default gcc 4.8.5 compiler that comes with the base Centos 7.6 OS. 

### Steps to reproduce the issue
Create an Azure HB60rs AMD instance, then: 
```console
$ git clone https://github.com/spack/spack.git
$./spack -d install -j 1 gcc@8.2.0

The build will fail with "illegal instruction" while building the perl dependency. 

### Platform and user environment

[mk2@spackhb bin]$ ./spack arch
linux-centos7-excavator
[mk2@spackhb bin]$ uname -a
Linux spackhb 3.10.0-957.21.3.el7.x86_64 #1 SMP Tue Jun 18 16:35:19 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux

[mk2@spackhb bin]$ cat /etc/redhat-release
CentOS Linux release 7.6.1810 (Core)

processor       : 59
vendor_id       : AuthenticAMD
cpu family      : 23
model           : 1
model name      : AMD EPYC 7551 32-Core Processor
stepping        : 2
microcode       : 0xffffffff
cpu MHz         : 1996.279
cache size      : 512 KB
physical id     : 1
siblings        : 28
core id         : 27
cpu cores       : 28
apicid          : 59
initial apicid  : 59
fpu             : yes
fpu_exception   : yes
cpuid level     : 13
wp              : yes
flags           : fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush mmx fxsr sse sse2 ht syscall nx mmxext fxsr_opt pdpe1gb rdtscp lm art rep_good nopl extd_apicid eagerfpu pni pclmulqdq ssse3 fma cx16 sse4_1 sse4_2 movbe popcnt aes xsave avx f16c rdrand hypervisor lahf_lm cmp_legacy cr8_legacy abm sse4a misalignsse 3dnowprefetch osvw retpoline_amd ssbd vmmcall fsgsbase bmi1 avx2 smep bmi2 rdseed adx smap clflushopt xsaveopt xsavec arat
bogomips        : 3992.95
TLB size        : 2560 4K pages
clflush size    : 64
cache_alignment : 64
address sizes   : 44 bits physical, 48 bits virtual

### Additional information

[mk2@spackhb bin]$ ./spack arch
linux-centos7-excavator
[mk2@spackhb bin]$ ./spack spec gcc@8.2.0
Input spec
--------------------------------
gcc@8.2.0

Concretized
--------------------------------
==> Warning: CROSS has not been set, however cannot detect scheduler.
gcc@8.2.0%gcc@4.8.5~binutils languages=c,c++,fortran ~nvptx~piclibs~strip arch=linux-centos7-steamroller
    ^gmp@6.1.2%gcc@4.8.5 arch=linux-centos7-steamroller
        ^autoconf@2.69%gcc@4.8.5 arch=linux-centos7-steamroller
            ^m4@1.4.18%gcc@4.8.5 patches=3877ab548f88597ab2327a2230ee048d2d07ace1062efe81fc92e91b7f39cd00,fc9b61654a3ba1a8d6cd78ce087e7c96366c290bc8d2c299f09828d793b853c8 +sigsegv arch=linux-centos7-steamroller
                ^libsigsegv@2.11%gcc@4.8.5 arch=linux-centos7-steamroller
            ^perl@5.26.2%gcc@4.8.5+cpanm patches=0eac10ed90aeb0459ad8851f88081d439a4e41978e586ec743069e8b059370ac +shared+threads arch=linux-centos7-steamroller
                ^gdbm@1.18.1%gcc@4.8.5 arch=linux-centos7-steamroller
                    ^readline@7.0%gcc@4.8.5 arch=linux-centos7-steamroller
                        ^ncurses@6.1%gcc@4.8.5~symlinks~termlib arch=linux-centos7-steamroller
                            ^pkgconf@1.6.1%gcc@4.8.5 arch=linux-centos7-steamroller
        ^automake@1.16.1%gcc@4.8.5 arch=linux-centos7-steamroller
        ^libtool@2.4.6%gcc@4.8.5 arch=linux-centos7-steamroller
    ^isl@0.18%gcc@4.8.5 arch=linux-centos7-steamroller
    ^mpc@1.1.0%gcc@4.8.5 arch=linux-centos7-steamroller
        ^mpfr@3.1.6%gcc@4.8.5 patches=66a5d58364113a21405fc53f4a48f4e8 arch=linux-centos7-steamroller
    ^zlib@1.2.11%gcc@4.8.5+optimize+pic+shared arch=linux-centos7-steamroller

[mk2@spackhb bin]$ ./spack compilers
==> Available compilers
-- gcc centos7-x86_64 -------------------------------------------
gcc@4.8.5


And here is the actual error: 

[mk2@spackhb bin]$ ./spack install gcc@8.2.0
==> Installing libsigsegv
==> Searching for binary cache of libsigsegv
==> Warning: No Spack mirrors are currently configured
==> No binary for libsigsegv found: installing from source
==> Fetching https://ftpmirror.gnu.org/libsigsegv/libsigsegv-2.11.tar.gz
######################################################################## 100.0%
==> Staging archive: /tmp/mk2/spack-stage/libsigsegv-2.11-56uvvg3i3cwe4nuexwa4ob2oabskyvgr/libsigsegv-2.11.tar.gz
==> Created stage in /tmp/mk2/spack-stage/libsigsegv-2.11-56uvvg3i3cwe4nuexwa4ob2oabskyvgr
==> No patches needed for libsigsegv
==> Building libsigsegv [AutotoolsPackage]
==> Executing phase: 'autoreconf'
==> Executing phase: 'configure'
==> Executing phase: 'build'
==> Executing phase: 'install'
==> Successfully installed libsigsegv
  Fetch: 2.06s.  Build: 6.37s.  Total: 8.44s.
[+] /home/mk2/spack/opt/spack/linux-centos7-steamroller/gcc-4.8.5/libsigsegv-2.11-56uvvg3i3cwe4nuexwa4ob2oabskyvgr
==> Installing m4
==> Searching for binary cache of m4
==> Warning: No Spack mirrors are currently configured
==> No binary for m4 found: installing from source
==> Fetching https://ftpmirror.gnu.org/m4/m4-1.4.18.tar.gz
######################################################################## 100.0%
==> Fetching https://src.fedoraproject.org/rpms/m4/raw/5d147168d4b93f38a4833f5dd1d650ad88af5a8a/f/m4-1.4.18-glibc-change-work-around.patch
######################################################################## 100.0%
==> Staging archive: /tmp/mk2/spack-stage/m4-1.4.18-sbitoczwdj6hphbucitkivqzengdkwgl/m4-1.4.18.tar.gz
==> Created stage in /tmp/mk2/spack-stage/m4-1.4.18-sbitoczwdj6hphbucitkivqzengdkwgl
==> Applied patch /home/mk2/spack/var/spack/repos/builtin/packages/m4/gnulib-pgi.patch
==> Applied patch https://src.fedoraproject.org/rpms/m4/raw/5d147168d4b93f38a4833f5dd1d650ad88af5a8a/f/m4-1.4.18-glibc-change-work-around.patch
==> Building m4 [AutotoolsPackage]
==> Executing phase: 'autoreconf'
==> Executing phase: 'configure'
==> Executing phase: 'build'
==> Executing phase: 'install'
==> Successfully installed m4
  Fetch: 2.69s.  Build: 36.25s.  Total: 38.95s.
[+] /home/mk2/spack/opt/spack/linux-centos7-steamroller/gcc-4.8.5/m4-1.4.18-sbitoczwdj6hphbucitkivqzengdkwgl
==> Installing pkgconf
==> Searching for binary cache of pkgconf
==> Warning: No Spack mirrors are currently configured
==> No binary for pkgconf found: installing from source
==> Fetching http://distfiles.dereferenced.org/pkgconf/pkgconf-1.6.1.tar.xz
######################################################################## 100.0%
==> Staging archive: /tmp/mk2/spack-stage/pkgconf-1.6.1-og7atogyyybk3h77fbmqzs6ijtnmk7rs/pkgconf-1.6.1.tar.xz
==> Created stage in /tmp/mk2/spack-stage/pkgconf-1.6.1-og7atogyyybk3h77fbmqzs6ijtnmk7rs
==> No patches needed for pkgconf
==> Building pkgconf [AutotoolsPackage]
==> Executing phase: 'autoreconf'
==> Executing phase: 'configure'
==> Executing phase: 'build'
==> Executing phase: 'install'
==> Successfully installed pkgconf
  Fetch: 0.85s.  Build: 6.09s.  Total: 6.94s.
[+] /home/mk2/spack/opt/spack/linux-centos7-steamroller/gcc-4.8.5/pkgconf-1.6.1-og7atogyyybk3h77fbmqzs6ijtnmk7rs
==> Installing ncurses
==> Searching for binary cache of ncurses
==> Warning: No Spack mirrors are currently configured
==> No binary for ncurses found: installing from source
==> Fetching https://ftpmirror.gnu.org/ncurses/ncurses-6.1.tar.gz
######################################################################## 100.0%
==> Staging archive: /tmp/mk2/spack-stage/ncurses-6.1-dsl2tvvsfmthlouvs3hfry22bkhbgbmi/ncurses-6.1.tar.gz
==> Created stage in /tmp/mk2/spack-stage/ncurses-6.1-dsl2tvvsfmthlouvs3hfry22bkhbgbmi
==> No patches needed for ncurses
==> Building ncurses [AutotoolsPackage]
==> Executing phase: 'autoreconf'
==> Executing phase: 'configure'
==> Executing phase: 'build'
==> Executing phase: 'install'
==> Successfully installed ncurses
  Fetch: 3.63s.  Build: 47.26s.  Total: 50.89s.
[+] /home/mk2/spack/opt/spack/linux-centos7-steamroller/gcc-4.8.5/ncurses-6.1-dsl2tvvsfmthlouvs3hfry22bkhbgbmi
==> Installing readline
==> Searching for binary cache of readline
==> Warning: No Spack mirrors are currently configured
==> No binary for readline found: installing from source
==> Fetching https://ftpmirror.gnu.org/readline/readline-7.0.tar.gz
######################################################################## 100.0%
==> Staging archive: /tmp/mk2/spack-stage/readline-7.0-flzuiozwhv6lpwn6r3mh36mzn5s33cea/readline-7.0.tar.gz
==> Created stage in /tmp/mk2/spack-stage/readline-7.0-flzuiozwhv6lpwn6r3mh36mzn5s33cea
==> No patches needed for readline
==> Building readline [AutotoolsPackage]
==> Executing phase: 'autoreconf'
==> Executing phase: 'configure'
==> Executing phase: 'build'
==> Executing phase: 'install'
==> Successfully installed readline
  Fetch: 4.31s.  Build: 10.80s.  Total: 15.10s.
[+] /home/mk2/spack/opt/spack/linux-centos7-steamroller/gcc-4.8.5/readline-7.0-flzuiozwhv6lpwn6r3mh36mzn5s33cea
==> Installing gdbm
==> Searching for binary cache of gdbm
==> Warning: No Spack mirrors are currently configured
==> No binary for gdbm found: installing from source
==> Fetching https://ftpmirror.gnu.org/gdbm/gdbm-1.18.1.tar.gz
######################################################################## 100.0%
==> Staging archive: /tmp/mk2/spack-stage/gdbm-1.18.1-ktiozt6vlvm4hq2a762duei5w2fc2vi4/gdbm-1.18.1.tar.gz
==> Created stage in /tmp/mk2/spack-stage/gdbm-1.18.1-ktiozt6vlvm4hq2a762duei5w2fc2vi4
==> No patches needed for gdbm
==> Building gdbm [AutotoolsPackage]
==> Executing phase: 'autoreconf'
==> Executing phase: 'configure'
==> Executing phase: 'build'
==> Executing phase: 'install'
==> Successfully installed gdbm
  Fetch: 2.07s.  Build: 9.22s.  Total: 11.29s.
[+] /home/mk2/spack/opt/spack/linux-centos7-steamroller/gcc-4.8.5/gdbm-1.18.1-ktiozt6vlvm4hq2a762duei5w2fc2vi4
==> Installing perl
==> Searching for binary cache of perl
==> Warning: No Spack mirrors are currently configured
==> No binary for perl found: installing from source
==> Fetching http://www.cpan.org/src/5.0/perl-5.26.2.tar.gz
######################################################################## 100.0%
==> Fetching http://search.cpan.org/CPAN/authors/id/M/MI/MIYAGAWA/App-cpanminus-1.7042.tar.gz
######################################################################## 100.0%
==> Fetching https://src.fedoraproject.org/rpms/perl/raw/004cea3a67df42e92ffdf4e9ac36d47a3c6a05a4/f/perl-5.26.1-guard_old_libcrypt_fix.patch
######################################################################## 100.0%
==> Staging archive: /tmp/mk2/spack-stage/perl-5.26.2-ljoae2gfrzfgvvdd4dx3vdputhi3qp3d/perl-5.26.2.tar.gz
==> Created stage in /tmp/mk2/spack-stage/perl-5.26.2-ljoae2gfrzfgvvdd4dx3vdputhi3qp3d
==> Staging archive: /tmp/mk2/spack-stage/resource-cpanm-ljoae2gfrzfgvvdd4dx3vdputhi3qp3d/App-cpanminus-1.7042.tar.gz
==> Created stage in /tmp/mk2/spack-stage/resource-cpanm-ljoae2gfrzfgvvdd4dx3vdputhi3qp3d
==> Moving resource stage
        source : /tmp/mk2/spack-stage/resource-cpanm-ljoae2gfrzfgvvdd4dx3vdputhi3qp3d/spack-src/
        destination : /tmp/mk2/spack-stage/perl-5.26.2-ljoae2gfrzfgvvdd4dx3vdputhi3qp3d/spack-src/cpanm/cpanm
==> Applied patch https://src.fedoraproject.org/rpms/perl/raw/004cea3a67df42e92ffdf4e9ac36d47a3c6a05a4/f/perl-5.26.1-guard_old_libcrypt_fix.patch
==> Building perl [Package]
==> Executing phase: 'configure'
==> Executing phase: 'build'
==> Error: ProcessError: Command exited with status 2:
    'make' '-j16'

1 error found in build log:
     939    cc -c -DPERL_CORE -D_REENTRANT -D_GNU_SOURCE -DAPPLLIB_EXP="/home/m
            k2/spack/opt/spack/linux-centos7-steamroller/gcc-4.8.5/perl-5.26.2-
            ljoae2gfrzfgvvdd4dx3vdputhi3qp3d/lib/perl5" -fwrapv -fno-strict-ali
            asing -pipe -fstack-protector-strong -I/home/mk2/spack/opt/spack/li
            nux-centos7-steamroller/gcc-4.8.5/gdbm-1.18.1-ktiozt6vlvm4hq2a762du
            ei5w2fc2vi4/include -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64 -D_F
            ORTIFY_SOURCE=2 -std=c89 -O2 -Wall -Werror=declaration-after-statem
            ent -Wextra -Wc++-compat -Wwrite-strings -fPIC op.c
     940    cc -fstack-protector-strong -L/home/mk2/spack/opt/spack/linux-cento
            s7-steamroller/gcc-4.8.5/gdbm-1.18.1-ktiozt6vlvm4hq2a762duei5w2fc2v
            i4/lib -o miniperl \
     941        opmini.o perlmini.o  gv.o toke.o perly.o pad.o regcomp.o dump.o
             util.o mg.o reentr.o mro_core.o keywords.o hv.o av.o run.o pp_hot.
            o sv.o pp.o scope.o pp_ctl.o pp_sys.o doop.o doio.o regexec.o utf8.
            o taint.o deb.o universal.o globals.o perlio.o perlapi.o numeric.o
            mathoms.o locale.o pp_pack.o pp_sort.o caretx.o dquote.o time64.o
            miniperlmain.o  -lpthread -lnsl -ldl -lm -lcrypt -lutil -lc
     942    LD_LIBRARY_PATH=/tmp/mk2/spack-stage/perl-5.26.2-ljoae2gfrzfgvvdd4d
            x3vdputhi3qp3d/spack-src ./miniperl -w -Ilib -Idist/Exporter/lib -M
            Exporter -e '<?>' || sh -c 'echo >&2 Failed to build miniperl.  Ple
            ase run make minitest; exit 1'
     943    /bin/sh: line 1: 55501 Illegal instruction     LD_LIBRARY_PATH=/tmp
            /mk2/spack-stage/perl-5.26.2-ljoae2gfrzfgvvdd4dx3vdputhi3qp3d/spack
            -src ./miniperl -w -Ilib -Idist/Exporter/lib -MExporter -e '<?>'
     944    Failed to build miniperl. Please run make minitest
  >> 945    make: *** [lib/buildcustomize.pl] Error 1
     946    make: *** Waiting for unfinished jobs....

See build log for details:
  /tmp/mk2/spack-stage/perl-5.26.2-ljoae2gfrzfgvvdd4dx3vdputhi3qp3d/spack-build-out.txt

[mk2@spackhb spack-src]$ ./miniperl
Illegal instruction
[mk2@spackhb spack-src]$ gdb ./miniperl
GNU gdb (GDB) Red Hat Enterprise Linux 7.6.1-114.el7
Copyright (C) 2013 Free Software Foundation, Inc.
License GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html>
This is free software: you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.  Type "show copying"
and "show warranty" for details.
This GDB was configured as "x86_64-redhat-linux-gnu".
For bug reporting instructions, please see:
<http://www.gnu.org/software/gdb/bugs/>...
Reading symbols from /tmp/mk2/spack-stage/perl-5.26.2-ljoae2gfrzfgvvdd4dx3vdputhi3qp3d/spack-src/miniperl...(no debugging symbols found)...done.
(gdb) run
Starting program: /tmp/mk2/spack-stage/perl-5.26.2-ljoae2gfrzfgvvdd4dx3vdputhi3qp3d/spack-src/./miniperl
[Thread debugging using libthread_db enabled]
Using host libthread_db library "/lib64/libthread_db.so.1".

Program received signal SIGILL, Illegal instruction.
0x000000000053c490 in Perl_new_ctype ()
Missing separate debuginfos, use: debuginfo-install glibc-2.17-260.el7_6.6.x86_64 nss-softokn-freebl-3.36.0-5.el7_5.x86_64


0x53c490 <Perl_new_ctype+1320>  bextr  $0x1d03,%rax,%rcx                                                                             │
   │0x53c499 <Perl_new_ctype+1329>  xor    %edx,%edx                                                                                     │
   │0x53c49b <Perl_new_ctype+1331>  test   $0x4,%al                                                                                      │
   │0x53c49d <Perl_new_ctype+1333>  rep movsq %ds:(%rsi),%es:(%rdi)                                                                      │
   │0x53c4a0 <Perl_new_ctype+1336>  jne    0x53c4d0 <Perl_new_ctype+1384>                                                                │
   │0x53c4a2 <Perl_new_ctype+1338>  test   $0x2,%al                                                                                      │
   │0x53c4a4 <Perl_new_ctype+1340>  jne    0x53c4ba <Perl_new_ctype+1362>                                                                │
   │0x53c4a6 <Perl_new_ctype+1342>  test   $0x1,%al                                                                                      │
   │0x53c4a8 <Perl_new_ctype+1344>  je     0x53c1e2 <Perl_new_ctype+634>                                                                 │
   │0x53c4ae <Perl_new_ctype+1350>  movzbl (%rsi,%rdx,1),%eax                                                                            │
   │0x53c4b2 <Perl_new_ctype+1354>  mov    %al,(%rdi,%rdx,1)                                                                             │
   │0x53c4b5 <Perl_new_ctype+1357>  jmpq   0x53c1e2 <Perl_new_ctype+634>                                                                 │
   │0x53c4ba <Perl_new_ctype+1362>  movzwl (%rsi,%rdx,1),%ecx                                                                            │
   │0x53c4be <Perl_new_ctype+1366>  mov    %cx,(%rdi,%rdx,1)                                                                             │
   │0x53c4c2 <Perl_new_ctype+1370>  add    $0x2,%rdx                                                                                     │
   │0x53c4c6 <Perl_new_ctype+1374>  test   $0x1,%al                                                                                      │
   │0x53c4c8 <Perl_new_ctype+1376>  je     0x53c1e2 <Perl_new_ctype+634>                                                                 │
   │0x53c4ce <Perl_new_ctype+1382>  jmp    0x53c4ae <Perl_new_ctype+1350>                                                                │
   │0x53c4d0 <Perl_new_ctype+1384>  mov    (%rsi),%edx                                                                                   │
   │0x53c4d2 <Perl_new_ctype+1386>  test   $0x2,%al                                                                                      │
   │0x53c4d4 <Perl_new_ctype+1388>  mov    %edx,(%rdi)                                                                                   │
   │0x53c4d6 <Perl_new_ctype+1390>  mov    $0x4,%edx                                                                                     │
   │0x53c4db <Perl_new_ctype+1395>  je     0x53c4a6 <Perl_new_ctype+1342>                                                                │
   │0x53c4dd <Perl_new_ctype+1397>  jmp    0x53c4ba <Perl_new_ctype+1362>                                                                │
   │0x53c4df <Perl_new_ctype+1399>  xor    %edx,%edx                                                                                     │
   │0x53c4e1 <Perl_new_ctype+1401>  mov    $0x1,%esi                                                                                     │
   │0x53c4e6 <Perl_new_ctype+1406>  mov    %rbx,%rdi                                                                                     │
   │0x53c4e9 <Perl_new_ctype+1409>  callq  0x53bec0 <Perl__is_in_locale_category>                                                        │
   │0x53c4ee <Perl_new_ctype+1414>  test   %al,%al                                                                                       │
   │0x53c4f0 <Perl_new_ctype+1416>  jne    0x53c2fe <Perl_new_ctype+918>                                                                 │
   │0x53c4f6 <Perl_new_ctype+1422>  jmpq   0x53c2cb <Perl_new_ctype+867>                                                                 │
   │0x53c4fb <Perl_new_ctype+1427>  test   $0x10,%al                                                                                     │
   │0x53c4fd <Perl_new_ctype+1429>  nopl   (%rax)                                                                                        │   └─────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┘
multi-thre Thread 0x7ffff In: Perl_new_ctype                                                                      Line: ??   PC: 0x53c490
(gdb)

--- a/lib/spack/llnl/util/cpu/microarchitectures.json
+++ b/lib/spack/llnl/util/cpu/microarchitectures.json
@@ -514,19 +514,8 @@
         }
       }
     },
-    "barcelona": {
-      "from": "x86_64",
-      "vendor": "AuthenticAMD",
-      "features": [
-        "mmx",
-        "sse",
-        "sse2",
-        "sse4a",
-        "abm"
-      ]
-    },
     "bulldozer": {
-      "from": "barcelona",
+      "from": "x86_64",
       "vendor": "AuthenticAMD",
       "features": [
         "mmx",
@@ -562,6 +551,8 @@
         "sse4a",
         "abm",
         "avx",
+        "xop",
+        "lwp",
         "aes",
         "pclmulqdq",
         "cx16",
@@ -570,7 +561,8 @@
         "sse4_2",
         "bmi1",
         "f16c",
-        "fma"
+        "fma",
+        "tbm"
       ],
       "compilers": {
         "gcc": {
@@ -590,6 +582,8 @@
         "sse4a",
         "abm",
         "avx",
+        "xop",
+        "lwp",
         "aes",
         "pclmulqdq",
         "cx16",
@@ -599,7 +593,8 @@
         "bmi1",
         "f16c",
         "fma",
-        "fsgsbase"
+        "fsgsbase",
+        "tbm"
       ],
       "compilers": {
         "gcc": {
@@ -619,6 +614,8 @@
         "sse4a",
         "abm",
         "avx",
+        "xop",
+        "lwp",
         "aes",
         "pclmulqdq",
         "cx16",
@@ -631,7 +628,8 @@
         "fsgsbase",
         "bmi2",
         "avx2",
-        "movbe"
+        "movbe",
+        "tbm"
       ],
       "compilers": {
         "gcc": {

--- a/lib/spack/llnl/util/cpu/microarchitectures.json
+++ b/lib/spack/llnl/util/cpu/microarchitectures.json
@@ -642,7 +642,7 @@
       }
     },
     "zen": {
-      "from": "excavator",
+      "from": "x86_64",
       "vendor": "AuthenticAMD",
       "features": [
         "bmi1",

--- a/lib/spack/spack/test/data/targets/linux-rhel6-piledriver
+++ b/lib/spack/spack/test/data/targets/linux-rhel6-piledriver
@@ -1,0 +1,29 @@
+processor       : 0
+vendor_id       : AuthenticAMD
+cpu family      : 21
+model           : 2
+model name      : AMD FX(tm)-8350 Eight-Core Processor
+stepping        : 0
+microcode       : 0x6000832
+cpu MHz         : 4013.832
+cache size      : 2048 KB
+physical id     : 0
+siblings        : 8
+core id         : 0
+cpu cores       : 4
+apicid          : 16
+initial apicid  : 0
+fpu             : yes
+fpu_exception   : yes
+cpuid level     : 13
+wp              : yes
+flags           : fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush mmx fxsr sse sse2 ht syscall nx mmxext fxsr_opt pdpe1gb rdtscp lm constant_tsc rep_good nopl nonstop_tsc extd_apicid aperfmperf pni pclmulqdq monitor ssse3 fma cx16 sse4_1 sse4_2 popcnt aes xsave avx f16c lahf_lm cmp_legacy svm extapic cr8_legacy abm sse4a misalignsse 3dnowprefetch osvw ibs xop skinit wdt lwp fma4 tce nodeid_msr tbm topoext perfctr_core perfctr_nb arat cpb hw_pstate npt lbrv svm_lock nrip_save tsc_scale vmcb_clean flushbyasid decodeassists pausefilter pfthreshold vmmcall bmi1
+bugs            : fxsave_leak sysret_ss_attrs
+bogomips        : 8030.15
+TLB size        : 1536 4K pages
+clflush size    : 64
+cache_alignment : 64
+address sizes   : 48 bits physical, 48 bits virtual
+power management: ts ttp tm 100mhzsteps hwpstate cpb eff_freq_ro
+
+# /proc/cpuinfo taken from https://www.redhat.com/archives/vfio-users/2015-August/msg00111.html

--- a/lib/spack/spack/test/llnl/util/cpu.py
+++ b/lib/spack/spack/test/llnl/util/cpu.py
@@ -113,7 +113,7 @@ def test_equality(supported_target):
     # Test microarchitectures that are ordered with respect to each other
     ('x86_64 < skylake', True),
     ('icelake > skylake', True),
-    ('piledriver <= zen', True),
+    ('piledriver <= steamroller', True),
     ('zen2 >= zen', True),
     ('zen >= zen', True),
     # Test unrelated microarchitectures

--- a/lib/spack/spack/test/llnl/util/cpu.py
+++ b/lib/spack/spack/test/llnl/util/cpu.py
@@ -26,6 +26,7 @@ from llnl.util.cpu import Microarchitecture  # noqa
     'linux-rhel7-ivybridge',
     'linux-rhel7-haswell',
     'linux-rhel7-zen',
+    'linux-rhel6-piledriver',
     'linux-centos7-power8le',
     'darwin-mojave-ivybridge',
     'darwin-mojave-broadwell',

